dbNSFP.pm

```diff
@@ -386,28 +402,28 @@ sub run {
     %$data = %$tmp_data;
 
     # convert data with multiple transcript values
-    # if($data->{Ensembl_transcriptid} =~ m/\;/) {
-
-    #   # find the "index" of this transcript
-    #   my @tr_ids = split(';', $data->{Ensembl_transcriptid});
-    #   my $tr_index;
-
-    #   for my $i(0..$#tr_ids) {
-    #     $tr_index = $i;
-    #     last if $tr_ids[$tr_index] =~ /^$tr_id(\.\d+)?$/;
-    #   }
-
-    #   next unless defined($tr_index);
-
-    #   # now alter other fields
-    #   foreach my $key(keys %$data) {
-    #     if($data->{$key} =~ m/\;/) {
-    #       my @split = split(';', $data->{$key});
-    #       die("ERROR: Transcript index out of range") if $tr_index > $#split;
-    #       $data->{$key} = $split[$tr_index];
-    #     } 
-    #   }
-    # }
+    if($self->{transcript_match} && $data->{Ensembl_transcriptid} =~ m/\;/) {
+
+      # find the "index" of this transcript
+      my @tr_ids = split(';', $data->{Ensembl_transcriptid});
+      my $tr_index;
+
+      for my $i(0..$#tr_ids) {
+        $tr_index = $i;
+        last if $tr_ids[$tr_index] =~ /^$tr_id(\.\d+)?$/;
+      }
+
+      next unless defined($tr_index);
```

Comment:
isn't tr_index always defined here? with value $#tr_ids if there was no matching transcript
instead you need above something like:
      for my $i(0..$#tr_ids) {
        if ($tr_ids[$tr_index] =~ /^$tr_id(\.\d+)?$/) {
           $tr_index = $i;
           last;
       }
    }

Reply:
This is correct yes, I'll change as you suggest.